### PR TITLE
fix(core): Handle wallet deletion failure for creation failure

### PR DESCRIPTION
**Pull request by amanCypherock**
https://app.clickup.com/t/9002019994/PRF-6377

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6377 Delete flow not triggered automatically after wallet creation failure or says to the user to delete wallet after flow exit](https://app.clickup.com/t/9002019994/PRF-6377)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6377 Delete flow not triggered automatically after wallet creation failure or says to the user to delete wallet after flow exit](https://app.clickup.com/t/9002019994/PRF-6377)

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6377 Delete flow not triggered automatically after wallet creation failure or says to the user to delete wallet after flow exit](https://app.clickup.com/t/9002019994/PRF-6377)

File: src/menu/create_wallet_menu.c [hunk 1]
```diff
@@ -161,14 +173,17 @@ static void create_wallet_menu_initialize(engine_ctx_t *ctx,
...
                                        ui_event_t ui_event,
                                        const void *data_ptr) {
+  bool result = 0xFF;    // default value
+  char wallet_name[NAME_SIZE] = "";
+
   if (UI_EVENT_LIST_CHOICE == ui_event.event_type) {
     switch (ui_event.list_selection) {
       case GENERATE_NEW_WALLET: {
-        create_new_wallet_flow();
+        result = create_new_wallet_flow(wallet_name);
```
Comment (ujjwal-cyph) [inline src/menu/create_wallet_menu.c:182]: take the wallet reference as the return of the flow. let's keep the flow trigger without any parameters

**Review by ujjwal-cyph — CHANGES_REQUESTED** [review]
change according to review comments

File: src/menu/create_wallet_menu.c (hunk 1 above)
Comment (amanCypherock) [reply src/menu/create_wallet_menu.c]: updated, re-review

File: src/menu/create_wallet_menu.c [hunk 2]
```diff
@@ -179,13 +193,36 @@ static void create_wallet_menu_handler(engine_ctx_t *ctx,
...
+    return;
+  }
+
+  message_scr_init(ui_text_creation_failed_delete_wallet);
+  if (0 != get_state_on_confirm_scr(0, 1, 2)) {
+    return;
+  }
+
+  delete_wallet_flow(flash_wallet);
+  flash_wallet = NULL;
```
Comment (ujjwal-cyph) [inline src/menu/create_wallet_menu.c:223]: local variable reset is not required.

File: src/menu/create_wallet_menu.c [hunk 3]
```diff
@@ -179,13 +193,36 @@ static void create_wallet_menu_handler(engine_ctx_t *ctx,
...
    * detected, or a create wallet flow was executed */
   engine_delete_current_flow_step(ctx);
 
   return;
 }
 
+void handle_wallet_creation_failure(Flash_Wallet *flash_wallet) {
+  if (NULL == flash_wallet) {
+    return;
+  }
```
Comment (ujjwal-cyph) [inline src/menu/create_wallet_menu.c:211]: can remove this duplicate check. it is already checked before calling this function

File: src/menu/create_wallet_menu.c (hunk 2 above)
Comment (amanCypherock) [reply src/menu/create_wallet_menu.c]: removed

File: src/menu/create_wallet_menu.c (hunk 3 above)
Comment (amanCypherock) [reply src/menu/create_wallet_menu.c]: removed

**Review by ujjwal-cyph — APPROVED** [review]

**Pull request merged**